Assets/Plugins/CountlySDK/Services/LocationService.cs

```diff
@@ -0,0 +1,154 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Plugins.CountlySDK.Enums;
+using Plugins.CountlySDK.Helpers;
+using Plugins.CountlySDK.Models;
+using UnityEngine;
+
+namespace Plugins.CountlySDK.Services
+{
+    public class LocationService
+    {
+        internal bool IsLocationDisabled { get; private set; }
+        internal string City { get; private set; }
+        internal string Location { get; private set; }
+        internal string IPAddress { get; private set; }
+        internal string CountryCode { get; private set; }
+
+        private readonly ConsentCountlyService _consentService;
+        private readonly RequestCountlyHelper _requestCountlyHelper;
+        private readonly CountlyConfiguration _countlyConfiguration;
+        
+
+        internal LocationService(CountlyConfiguration countlyConfiguration, RequestCountlyHelper requestCountlyHelper)
+        {
+            _countlyConfiguration = countlyConfiguration;
+            _requestCountlyHelper = requestCountlyHelper;
+
+            if (countlyConfiguration.IsLocationDisabled)
+            {
+                City = null;
+                Location = null;
+                IPAddress = null;
+                CountryCode = null;
+                IsLocationDisabled = countlyConfiguration.IsLocationDisabled;
+            }
+            else
+            {
+                City = countlyConfiguration.City;
+                Location = countlyConfiguration.Location;
+                IPAddress = countlyConfiguration.IPAddress;
+                CountryCode = countlyConfiguration.CountryCode;
+                IsLocationDisabled = countlyConfiguration.IsLocationDisabled;
+            }
+        }
+
+        internal async Task SendRequestWithEmptyLocation()
+        {
+            var requestParams =
+               new Dictionary<string, object>();
+
+            requestParams.Add("location", string.Empty);
+
+            await _requestCountlyHelper.GetResponseAsync(requestParams);
+        }
+
+        internal async Task SendIndependantLocationRequest()
```

Comment:
This functions logic is false as it doesn't check if any valid values are added and would send just the base request when it is called.

Reply:
In SendIndependantLocationRequest, you asking me to valide IP address , country code, city and location before sending to server?

Reply:
I'm not sure what kind of validation are you thinking of. 

This call will work wrong when no session consent is given and no init time location info was provided

Reply:
This functions logic is false as it doesn't check if any valid values are added,

you are saying that al least 1 value should add in request other wise don't send to server. right?